classes/Provider/Gateways/WebGatewayProvider.php

```diff
@@ -59,6 +60,12 @@ public function boot(Application $app)
             return SpeakerAccess::userHasAccess($app);
         };
 
+        $csrfChecker = function (Request $req) use ($app) {
+            $checker = $app[CsrfCheck::class];
+
+            return $checker->checkCsrf($req->get('token_id'), $req->get('token'));
```

Comment:
If we adjusted the `CsrfCheck` class as suggested above, we could modify the middleware to something like this:

```php
$csrfChecker = function(Request $request) use ($app) {
   $validator = $app[CsrfCheck::class];

    if (!$validator->isValid($request)) {
        return new RedirectResponse('/dashboard');
    }

    return $request;
}
```

This would have the advantage that the target of the redirection is not hard-coded into the validator, but hard-coded into the definition of the middleware, and it's more obvious that we either return a `RedirectResponse` here, or pass through the initial request.